### PR TITLE
Bug fix/Internal improvement: Update migrations

### PR DESCRIPTION
**Pull request by eggplantzzz**
Previously, useful stack traces were not showing when an error in migrations happened. This was due to the fact that when errors were caught in `migrations/index.js`, a new error was created with the caught error rather than throwing the error directly. This messed up the stack trace. This PR causes the code to throw that error directly.

This PR also removes the `async` library from `@truffle/migrate` and uses Promises in the place where that library was utilized for the migrations. The logic should remain the same.

### CODE REVIEW

File: packages/migrate/index.js [hunk 1]
```diff
@@ -81,15 +78,14 @@ const Migrate = {
...
-
       // Don't rerun the last completed migration.
       await this.runFrom(lastMigration + 1, options);
 
       if (callbackPassed) return callback();
       return;
     } catch (error) {
       if (callbackPassed) return callback(error);
-      throw new Error(error);
+      throw error;
```
[the remark this answers is not in the document]
Comment (CruzMolina) [reply packages/migrate/index.js]: Often I've also had to change this line as well: https://github.com/trufflesuite/truffle/blob/develop/packages/migrate/migration.js#L124

[the remark this answers is not in the document]
Comment (CruzMolina) [reply packages/migrate/index.js]: Other thing to note is I believe the `Error` output in both instances (in the linked code & current code being reviewed) is different when testing unbundled vs bundled.

There _might_ be a bundled reason to throw a new Error. Not sure, but maybe there's already tests for this.

[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/migrate/index.js]: Why didn't you put in a PR for this??

**Comment by eggplantzzz** [comment]
I'm going to close this and break this up into two PRs.
Bug fix PR --> https://github.com/trufflesuite/truffle/pull/2596

**Pull request closed without merging**